soc/arm/ambiq/apollo4x/soc.c

```diff
@@ -0,0 +1,21 @@
+/*
+ * Copyright (c) 2023 Antmicro <www.antmicro.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/arch/cpu.h>
+#include <zephyr/init.h>
+#include <zephyr/kernel.h>
+#include <zephyr/logging/log.h>
+
+#include <soc.h>
```

Comment:
soc.h is a header that exists purely because of how CMSIS is included by the ARM port, you should include what you need, ie, hal headers.

Reply:
Removed this include and added hal header.